dvc/remote/gdrive.py

```diff
@@ -126,120 +115,90 @@ def __init__(self, repo, config):
         )
 
         self._list_params = None
-        self._gdrive = None
 
-        self._cache_initialized = False
-        self._remote_root_id = None
-        self._cached_dirs = None
-        self._cached_ids = None
-
-    @property
-    @wrap_with(threading.RLock())
+    @wrap_prop(threading.RLock())
+    @cached_property
     def drive(self):
         from pydrive2.auth import RefreshError
+        from pydrive2.auth import GoogleAuth
+        from pydrive2.drive import GoogleDrive
+
+        if os.getenv(RemoteGDrive.GDRIVE_USER_CREDENTIALS_DATA):
+            with open(
+                self._gdrive_user_credentials_path, "w"
+            ) as credentials_file:
+                credentials_file.write(
+                    os.getenv(RemoteGDrive.GDRIVE_USER_CREDENTIALS_DATA)
+                )
 
-        if not self._gdrive:
-            from pydrive2.auth import GoogleAuth
-            from pydrive2.drive import GoogleDrive
+        GoogleAuth.DEFAULT_SETTINGS["client_config_backend"] = "settings"
```

Comment:
not directly related - would really love this to go away from the class ... is there a good way? or at least assign the whole dict at once to DEFAULT_SETTINGS? This way it is done now is quite ugly and bloated

Reply:
Yeah, it is. One can use `GoogleAuth.DEFAULT_SETTINGS.update()`, but the right way would be to make `GoogleAuth` accept settings as a dict not only file.